### ISSUE TITLE
rules to improve

### ISSUE THREAD

**Issue by duaraghav8** [duaraghav8/Solium#169]
- [ ] `indentation` - doesn't lint for indent inside the `assembly` block

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 120.0 DAI (120.0 USD @ $1.0/DAI) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/duaraghav8/Solium/169/780).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $22,450.94 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by nemaniarjun** [comment]
@duaraghav8  This and https://github.com/duaraghav8/Solium/issues/94 Seem very similar. @ceresstation
Can you both figure out how exactly these 2 issues will be done? As there seems to be (a possibility of) redundant work.

**Comment by gomesalexandre** [comment]
Indeed, they seem to overlap @nemaniarjun

**Comment by duaraghav8** [comment]
@nemaniarjun @gomesalexandre @aerophile there is no overlap. 1 issue exclusively focuses on the `fix()` function, which is written regardless of how its surrounding code looks like (of course, it needs some info about the solidity code to know what to replace/add/edit).

The other issue focuses on improvement in the sense that the rule itself is written very poorly (for which I am to blame). So it primarily requires refactoring and completing the set of checks the rule must actually be performing.

And if a check is not yet present, then you need not worry about implementing a fix for that. eg- if `indentation` doesn't check indent used by array items declared over multiple lines (at the time when you're developing fixes for that rule), then you need not write a fix for that and your PR will still be accepted.

Sorry for the confusion. Hope I cleared things now? Please let me know if there is any further confusion

**Comment by mkosowsk** [comment]
@writeprovidence with the previous comment, do you have the information and tools you need to solve this issue? :)

**Comment by writeprovidence** [comment]
@mkosowsk i think i do. will definately ask questions if need be

**Comment by mkosowsk** [comment]
@writeprovidence just a friendly check-in here 😀 have you had a chance to take a look at this issue?

**Comment by writeprovidence** [comment]
still trying to figure out something

**Comment by duaraghav8** [comment]
@writeprovidence let me know if it's something you aren't able to figure out from the docs

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 6 days, 2 hours from now.
Please review their action plans below:


1. [mridulnagpal](https://gitcoin.co/profile/mridulnagpal) has started work. 
    
    If this is still open I would love to work on this issue

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/duaraghav8/Solium/169/780).

**Comment by nemaniarjun** [comment]
@duaraghav8 Can you tell me which all rules are still open to work on? I finally have some time to do this!
:smile:

**Comment by duaraghav8** [comment]
Hey @nemaniarjun, sorry for replying so late. All rules are still open. There was only 1 PR opened a while ago but I it hasn't been worked on for a while so I'll favour another PR over it.

**Comment by oogetyboogety** [comment]
@duaraghav8 is there a special set of rules to indent for assembly blocks? or should they all be indented one level deeper at the same level in the assembly block?
Also, what should be done for the whitespace rules?

Thanks!

**Comment by duaraghav8** [comment]
ah, sorry I didn't write a proper description for whitespace rules.
currently, they're written very poorly - hacks put together through regular expressions.
what is needed primarily is the refactoring of whitespace rules such that they pass all existing tests, use ES6 and become much more maintainable (ie use good JS practices).

there's no separate rule for indenting assembly, all indentation-related checks are in a single rule file.

**Comment by oogetyboogety** [comment]
@duaraghav8 Thanks I'll take a look at refactoring the whitespace rule for the tests and try some ES6 best practices that I know of.

is there any specific assembly style indentation/style guide you would like to follow? or should all inline assembly be on the same column, unless there is a label to jump to or an additional block in the assembly, and that's it; an example of this is here:
https://github.com/androlo/solidity-workshop/blob/master/tutorials/2016-04-04-solidity-inline-assembly-I.md

**Comment by duaraghav8** [comment]
All on same column + indentation for labels or blocks (because that is more natural and intuitive for end user)

**Comment by PixelantDesign** [comment]
Hello duaraghav8! This is Alisa from Gitcoin. Is this issue still active?

**Comment by duaraghav8** [comment]
Hey Alisa, The issue on the project is still active.
I'm unsure of its status on Gitcoin. If it is open on Gitcoin, feel free to close it as I don't see anyone taking this up.

**Comment by obernardovieira** [comment]
I've actually looked at it a few days ago and asked myself if it was being developed.
I could give it a try, also because I want this. But unfortunately I can try it only after 12th June.

**Comment by raininja** [comment]
I can give this a whirl @duaraghav8, I'm already familiar with the codebase, what say you?

**Comment by raininja** [comment]
It's still open, can I take it?
@duaraghav8 
@PixelantDesign

**Comment by raininja** [comment]
Tried to "start work" but the bounty was taken by another user

**Comment by raininja** [comment]
So nothing is being done on this?

**Comment by duaraghav8** [comment]
@raininja I'm no longer sure whether the bounties are still valid or not. I can confirm that the issue is  still active @PixelantDesign but not sure if the bounty still applies or not.

**Comment by ceresstation** [comment]
Hi @raininja you are welcome to take this one on if you have time, I'll even add a tip. Just approved you.

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Work for 120.0 DAI (120.0 USD @ $1.0/DAI) has been submitted by__:

 1. [@vporton](https://gitcoin.co/vporton)



@ceresstation please take a look at the submitted work:
* [PR](https://github.com/duaraghav8/Ethlint/pull/278) by @vporton

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/duaraghav8/Solium/169/780)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/duaraghav8/Solium/169/780).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $80,075.60 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__The funding of 120.0 DAI (120.0 USD @ $1.0/DAI)  attached to this issue has been approved & issued to @vporton.__  

 
* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/duaraghav8/Solium/169/780)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $75,165.74 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
